TipCatDotNet.Api/Services/Payments/TransactionService.cs

```diff
@@ -0,0 +1,89 @@
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using CSharpFunctionalExtensions;
+using Stripe;
+using TipCatDotNet.Api.Data;
+using TipCatDotNet.Api.Data.Models.Payment;
+using TipCatDotNet.Api.Models.HospitalityFacilities;
+using TipCatDotNet.Api.Models.Payments;
+using HappyTravel.Money.Enums;
+using System.Linq;
+using System.Linq.Expressions;
+using Microsoft.EntityFrameworkCore;
+using TipCatDotNet.Api.Filters.Pagination;
+
+namespace TipCatDotNet.Api.Services.Payments
+{
+    public class TransactionService : ITransactionService
+    {
+        public TransactionService(AetherDbContext context)
+        {
+            _context = context;
+        }
+
+
+        public async Task<Result> Add(PaymentIntent paymentIntent, CancellationToken cancellationToken = default)
+        {
+            var now = DateTime.UtcNow;
+
+            var newTransaction = new Transaction
+            {
+                Amount = paymentIntent.Amount,
+                Currency = Enum.Parse<Currencies>(paymentIntent.Currency),
+                MemberId = int.Parse(paymentIntent.Metadata["MemberId"]),
+                PaymentIntentId = paymentIntent.Id,
+                State = paymentIntent.Status,
+                Created = paymentIntent.Created,
+                Modified = now,
+            };
+
+            _context.Transactions.Add(newTransaction);
+            await _context.SaveChangesAsync(cancellationToken);
+
+            return Result.Success();
+        }
+
+
+        public async Task<Result<List<TransactionResponse>>> Get(MemberContext context, PaginationFilter filter, CancellationToken cancellationToken = default)
+            => await _context.Transactions
+                .Where(t => t.MemberId == context.Id && t.State == "succeeded")
```

Comment:
It's not clear what the method returns only succeeded transactions. Add a comment for that